x-pack/plugins/spaces/server/plugin.ts

```diff
@@ -148,18 +148,30 @@ export class SpacesPlugin
       logger: this.log,
     });
 
-    const externalRouter = core.http.createRouter<SpacesRequestHandlerContext>();
```

Comment:
Is there a reason we instantiated independent external and internal routers?

Reply:
Maybe @legrego can remember...

Reply:
IIRC this design made sense in the original "new platform migration" for uninteresting reasons. It's no longer optimal though, so I have no objections to changing things around